.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,23 @@
+name: build and test gradle project
+
+on:
+  push:
+    branches-ignore:
+    - master
+    - version-*
+    - dependabot*
+    paths-ignore:
+      - README.md
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout project sources
+        uses: actions/checkout@v3
+      - name: Setup Gradle
+        uses: gradle/gradle-build-action@v2
+        with:
+          cache-read-only: false
+      - name: Run build with Gradle Wrapper
+        run: gradle clean build --info
```

Comment:
It does not look like you are using Gradle Wrapper here. Should we change the name?